### PR TITLE
fix(persons-modal): Correct series names display in persons modal

### PR DESCRIPTION
**Pull request by sumtzehern**
- Update the  function to map custom names () from the  to the dropdown options.
- Add checks to handle cases where  or  might be undefined, preventing potential errors.
- Ensure user-defined custom names are displayed in the series dropdown instead of the default  labels.

## Problem
The series dropdown within the persons modal was displaying the default `$pageview` label instead of the custom names provided by the user. This issue was caused by the dropdown not properly mapping the custom names from the `query.source.series`.

## Changes

- Modified the `cleanedInsightActorsQueryOptions` function to include custom names from the `query.source.series`.
- Added checks to handle cases where `query` or `query.source` might be undefined.
- Updated the dropdown to display custom names instead of the default `$pageview` labels.

image

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

- Ran `pnpm test:unit` to ensure all unit tests pass.
- Manually tested the changes by:
  1. Creating or viewing an insight with multiple pageviews as series.
  2. Renaming each series using the pencil icon in the Detailed results table.
  3. Hovering over a point on the chart to see the name changes in the modal.
  4. Clicking to view people and confirming the series dropdown displays the custom names.

### CODE REVIEW

File: frontend/src/scenes/trends/persons-modal/persons-modal-utils.tsx [hunk 1]
```diff
@@ -53,9 +53,30 @@ export const pathsTitle = (props: { mode: pathModes; label: string }): React.Rea
...
+                return [
+                    key,
+                    value.map((v: any, index: number) => ({
+                        ...v,
+                        label: seriesNames[index],
+                    })),
+                ]
+            }
+            return [key, value]
+        })
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/scenes/trends/persons-modal/persons-modal-utils.tsx:81]: Actually, let me push this change if you don't mind

File: frontend/src/scenes/trends/persons-modal/persons-modal-utils.tsx [hunk 2]
```diff
@@ -53,9 +55,29 @@ export const pathsTitle = (props: { mode: pathModes; label: string }): React.Rea
...
+    const seriesNames = isTrendsQuery(source) ? source.series.map((s: any) => s.custom_name) : []
+    const cleanedOptionsWithAdjustedSeriesNames: [string, any[]][] = cleanedOptions.map(([key, value]) => {
+        if (key === 'series') {
+            return [
+                key,
+                value.map((v: any, index: number) => ({
+                    ...v,
+                    label:
+                        seriesNames[index] ??
+                        getCoreFilterDefinition(v.label, TaxonomicFilterGroupType.Events)?.label ??
```
Comment (Twixes) [inline frontend/src/scenes/trends/persons-modal/persons-modal-utils.tsx:74]: An extra bit of missing formatting I thought is worth adding: We'll now also format PostHog core events (prefixed with $), e.g. "$pageview" becomes "Pageview"

**Review by Twixes — APPROVED** [review]
Thank you, that's a pleasant improvement!

**Comment by sumtzehern** [comment]
@Twixes , 
Great! good to know it resolves the issue. Thanks for reviewing the code and making these improvements.

**Comment by Twixes** [comment]
@sumtzehern As a thank you token, please accept a $50 gift card to [our merch store](https://merch.posthog.com). You should have it in your @vt.edu inbox that's attached to your commits. :)

**Comment by sumtzehern** [comment]
@Twixes, Just wanted to let you know that this fix for issue #23482 has been implemented. Would it be better to consider closing this issue?

**Comment by Twixes** [comment]
Of course, missed that this didn't auto-close the issue. Closed now.

**Pull request merged**